### DISCUSSION TITLE
Q: How can I get the text values of a node?

### DISCUSSION THREAD

**Discussion by droslean** [yuin/goldmark#296] [Q&A] [answered]
In https://pkg.go.dev/github.com/yuin/goldmark@v1.4.11/ast#Node

```go
type Node interface {
...
// Text returns text values of this node.
Text(source []byte) []byte
...
```

However, it is unclear what the `source` bytes are.

**Comment by droslean** [comment]
@yuin Any insights?

**Comment by droslean** [comment, marked as answer]
So it turns out that the `source` is the actual bytes that we passed to parser. It is unclear to me why we need to pass them again.

**Comment by shazib-summar** [comment]
@droslean hey thanks for the explanation. I had another question hoping you could help; can you please elaborate how some nodes have `Lines` func defined on them but some dont? How do I tell which node kinds have Lines and which ones dont?

Also whats the difference between Text Block, and Paragraph? Similary Text vs String? I went through the docs but didnt find anything helpful there.

Thanks in advance 🙏🏾
